drivers/amazonec2/amazonec2.go

```diff
@@ -1538,3 +1533,36 @@ func hasTagKey(tags []*ec2.Tag, key string) bool {
 	}
 	return false
 }
+
+func (d *Driver) generateBDMList() ([]*ec2.BlockDeviceMapping, error) {
+	var bdmList []*ec2.BlockDeviceMapping
+
+	images, err := d.getClient().DescribeImages(&ec2.DescribeImagesInput{
```

Comment:
this is already done in preCheck (https://github.com/rancher/machine/blob/2948c2b384ed70a09114c5b1e7d3a0fd14b5a987/drivers/amazonec2/amazonec2.go#L556-L581), think you can store it from there and use the same details here?

Reply:
Feedback incorporated.